### PR TITLE
update documentation chapter

### PR DESCRIPTION
**Pull request by SaranjeetKaur**
Please review the updated version of the documentation chapter of the dev guide.

### CODE REVIEW

File: 07-documenting_R.Rmd
Comment (nbenn) [reply 07-documenting_R.Rmd:13]:
```suggestion
This chapter is aimed at authors and potential authors of documentation for R. More specifically, it is for people contributing to the standard documentation and developing additional documents using the same tools as the standard documents. Any time you feel that you can clarify existing documentation or provide documentation that is missing, your contribution is welcome and appreciated. If you find it difficult to deal with the markup language, you can ask for help for that part too. Please do not let the material in this chapter stand between the documentation and your desire to help out. However, not every good faith effort to to change or extend the documentation will be accepted - sometimes the patch contribution might be incorrect, other times, while a change in wording may make some things clearer, other finer details of some corner case may become less clear, leading to the patch being declined or modified by the R-core member reviewing it before applying it (if they agree the issue is important enough to fix).
```

Comment (nbenn) [reply 07-documenting_R.Rmd:13]: Sorry, above my suggestion is a bit masked (I was hoping for some diff highlighting): I changed only the part "other finer details of some corner case may become less clear"

File: 07-documenting_R.Rmd
Comment (nbenn) [inline 07-documenting_R.Rmd:48]: If `\synopsis` was retired w/ R 3.1.0. I'd go ahead and drop mentioning this altogether.

File: 07-documenting_R.Rmd
Comment (nbenn) [inline 07-documenting_R.Rmd:17]: Just a suggestion: Instead of referring to an external resource, you could also copy a significant portion (or all) of that as a code block and then when explaining the components later on in a bit more detail, point to the sections in the example. This would make for a bit more self-contained reading.

File: 07-documenting_R.Rmd
Comment (nbenn) [inline 07-documenting_R.Rmd:81]: I feel this has gotten overly complicated. The relevant info is: you can directly search for docs issues on Bugzilla + maybe a link to such a view. Using Bugzilla teminology such as "component" and explaining that this is a piece of metadata attached to each bug report is besides the point. Just my 2 ¢...

File: 07-documenting_R.Rmd
Comment (hturner) [inline 07-documenting_R.Rmd:99]: I think we might want to move this to a new chapter. There are various websites that the R Core team maintain, each with their own way to report issues (R FAQ, Windows FAQ, Mac FAQ, R project web-site, CRAN website). These are all outside the scope of Bugzilla. This might be a new section in the Bug reporting chapter (since we also document there how to report bugs in non-R-Core maintained packages), or a small chapter on "R project websites". Perhaps the latter is useful as there are also other sites on the r-project domain, like translation.r-project,org and contributor.r-project.org that could be good to document briefly somewhere.

File: 07-documenting_R.Rmd
Comment (hturner) [inline 07-documenting_R.Rmd:103]: This should then be in the previous section about helping with documentation.

File: 07-documenting_R.Rmd
Comment (hturner) [inline 07-documenting_R.Rmd:108]: This part should be elevated to a new section. Since you already cite R-exts earlier, you can probably drop this See also section.

To make this guide more self-contained, you should briefly describe the bug you link to and use a hyperlink with a title of the linked bug: [Bug 15221 - R-admin/'Installing R under Windows': Missing argument name](https://bugs.r-project.org/show_bug.cgi?id=15221).

Also, you could add a list or table with the manual names and their nicknames.

"The [Texinfo manual](https://www.gnu.org/software/texinfo/manual/texinfo/texinfo.html) should be referred to for [how to mark up text]."

File: 07-documenting_R.Rmd
Comment (hturner) [inline 07-documenting_R.Rmd:109]: Again - use hyperlink with bug title vs URL.

Ideally, someone would be able to build the manuals after following our instructions to build R from source. This probably requires an update to the installation instructions first, then perhaps we can add simple instructions here for building the manuals (cc @llrs) - no need to wait for this before merging this PR.

**Review by hturner — COMMENTED** [review]
I think this chapter still needs a lot of work, so I'm afraid each of my comments combine several issues and will not be straightforward to resolve. Hopefully it is clear how to respond to them though.

File: 07-documenting_R.Rmd
Comment (llrs) [reply 07-documenting_R.Rmd:109]: I think that one doesn't need to build R from source to build the manuals, but I will check it.

File: 07-documenting_R.Rmd
Comment (bettinagruen) [reply 07-documenting_R.Rmd:17]: I agree that it would be good to mention roxygen and that this creates .Rd files but that this guide focuses on .Rd files only because the man pages for the base distribution are written directly as .Rd files.

File: 07-documenting_R.Rmd
Comment (bettinagruen) [reply 07-documenting_R.Rmd:99]: Would this also apply to the manuals?

**Review by bettinagruen — CHANGES_REQUESTED** [review]

**Pull request merged**
